### DISCUSSION TITLE
React Select giving object for selected option instead of value.

### DISCUSSION THREAD

**Discussion by azad-steadybiz** [react-hook-form/react-hook-form#2427] [Q&A]
I am using React hook form with React Select, so when I am selecting any value option I am getting the object of selected option instead of selected option value.

**Comment by bluebill1049** [comment]
Take a look at this CSB: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by azad-steadybiz** [reply]
I have checked this example, but I want the selected value to be in string not be in object.

**Reply by bluebill1049** [reply]
`object` is what react-select expects. options:

1. build a wrapper component which takes a string and receive a string
2. sort out at the submitting level.

**Reply by azad-steadybiz** [reply]
How to do it with react hook form, can you share sandbox

**Reply by bluebill1049** [reply]
1. `<Controller render={(props) => <SelectWrapper {...props}><ReactSelect /></SelectWrapper>} />`

inside `SelectWrapper` just take `string` and produce an object for react-select.

2. sort out the submission level just convert the `object` into `string`
